kite-morphlines/kite-morphlines-phonenumber/src/main/java/org/kitesdk/morphline/phonenumber/PhoneNumberBuilder.java

```diff
@@ -0,0 +1,166 @@
+		private final static String FIELD_DEFAULT_REGION_CODE = "defaultRegionCode";
+		private final static String FIELD_REGION_CODE_FIELD = "regionCodeField";
+		private final static String FIELD_OUTPUT_PHONE_FORMAT = "outputPhoneFormat";
+
+		private final static String DEFAULT_OUTPUT_FIELD_PREFIX = "phone_";
+		private final static String DEFAULT_REGION_CODE = "US";
+		private final static String DEFAULT_OUTPUT_PHONE_FORMAT = "INTERNATIONAL";
+
+		private final static String SUFFIX_PHONE = "phone";
+		private final static String SUFFIX_COUNTRY_CODE = "country_code";
+		private final static String SUFFIX_REGION_CODE = "region_code";
+
+		private final String inputField;
+		private final String outputPhoneField;
+		private final String outputCountryCodeField;
+		private final String outputRegionCodeField;
+		private final String defaultRegionCode;
+		private final String regionCodeField;
+		private final PhoneNumberUtil.PhoneNumberFormat phoneNumberFormat;
+		private final PhoneNumberUtil phoneNumberUtil;
+
+		public PhoneNumber(final CommandBuilder builder, final Config config, final Command parent, final Command child,
+				final MorphlineContext context) {
+			super(builder, config, parent, child, context);
+
+			this.phoneNumberUtil = PhoneNumberUtil.getInstance();
+			this.inputField = getConfigs().getString(config, FIELD_INPUT_FIELD);
+
+			final String outputFieldPrefix = getConfigs()
+					.getString(config, FIELD_OUTPUT_FIELD_PREFIX, DEFAULT_OUTPUT_FIELD_PREFIX);
+			this.outputPhoneField = outputFieldPrefix + SUFFIX_PHONE;
+			this.outputCountryCodeField = outputFieldPrefix + SUFFIX_COUNTRY_CODE;
+			this.outputRegionCodeField = outputFieldPrefix + SUFFIX_REGION_CODE;
+			this.regionCodeField = getConfigs().getString(config, FIELD_REGION_CODE_FIELD, null);
+
+			this.defaultRegionCode = getConfigs().getString(config, FIELD_DEFAULT_REGION_CODE, DEFAULT_REGION_CODE);
+			if (!this.phoneNumberUtil.getSupportedRegions().contains(this.defaultRegionCode)) {
+				throw new MorphlineCompilationException(
+						"The specified default region code is not valid: " + this.defaultRegionCode, config);
+			}
+
+			final String phoneNumberFormatString = getConfigs()
+					.getString(config, FIELD_OUTPUT_PHONE_FORMAT, DEFAULT_OUTPUT_PHONE_FORMAT)
+					.toUpperCase(Locale.ENGLISH);
+			try {
+				this.phoneNumberFormat = PhoneNumberUtil.PhoneNumberFormat.valueOf(phoneNumberFormatString);
+			} catch (IllegalArgumentException ex) {
+				throw new MorphlineCompilationException(
+						"The specified phone number format is not valid: " + phoneNumberFormatString, config);
+			}
+
+			validateArguments();
+		}
+
+		@Override
+		protected boolean doProcess(final Record record) {
+
+			final Object objValue = record.getFirstValue(inputField);
+			if (objValue == null) {
+				LOG.debug("Input field ({}) was null", inputField);
+				return false;
+			} else if (!(objValue instanceof String)) {
+				if (LOG.isWarnEnabled()) {
```

Comment:
LOG.isWarnEnabled() isn't really necessary here because it doesn't significantly help performance in this context.